Hash info missing for

### Acton Version

0.26.0.20260211.10.52

### Steps to Reproduce and Observed Behavior

I made the following YANG change to make use of the new container-transform attachment:
```bash
one-fabric-orch on  main [!?] via ↯ took 17s
❯ git diff
diff --git a/gen/yang/rfs/one-fabric-rfs.yang b/gen/yang/rfs/one-fabric-rfs.yang
index ab036f2..66c1574 100644
--- a/gen/yang/rfs/one-fabric-rfs.yang
+++ b/gen/yang/rfs/one-fabric-rfs.yang
@@ -11,9 +11,7 @@ module one-fabric-rfs {
     }

     augment "/orfs:rfs" {
-        // TODO: should really be a P-container
-        list base-config {
-            key device-name;
+        container base-config {

             orchestron:rfs-transform one_fabric.rfs.BaseConfig;
```

I triggered a re-gen:
```
one-fabric-orch on  main [!] via ↯
❯ make gen
cd gen && acton build  && out/bin/one_fabric_gen
Building project in /Users/klambrec/Developer/autlan/one-fabric-orch/gen
Resolving dependencies (fetching if missing)...
Using cached pkg dependency actmf (N-V-__8AAJtaBABOb7WyjNcnWG3420pPMwvehI80qJohWsmC)
Using cached pkg dependency orchestron (N-V-__8AAFBUEQBjXaW-92KIz_iNxo4_QeS9zuIj8QyUujZO)
Using cached pkg dependency yang (N-V-__8AAIOYIABUmXlnf1AZF-mtRvZcsWOi3MzODQhXHGxf)
Using cached pkg dependency netconf (N-V-__8AAOL3AQD7Oy0qj2nAbYZ0u3dGWSGfvqABuALB3K0y)
   Finished final compilation                                         0.786 s
Applying transform no_comments to Cisco-IOS-XR-policy-repository-cfg.yang.orig
Writing Cisco-IOS-XR-policy-repository-cfg.yang
Applying transform no_comments to Cisco-IOS-XR-um-if-l2-ethernet-cfg.yang.orig
Writing Cisco-IOS-XR-um-if-l2-ethernet-cfg.yang
Applying transform no_comments to Cisco-IOS-XR-um-mpls-ldp-cfg.yang.orig
Writing Cisco-IOS-XR-um-mpls-ldp-cfg.yang
Applying transform no_comments to Cisco-IOS-XR-um-router-bgp-cfg.yang.orig
Writing Cisco-IOS-XR-um-router-bgp-cfg.yang
Applying transform no_comments to Cisco-IOS-XR-um-router-isis-cfg.yang.orig
Writing Cisco-IOS-XR-um-router-isis-cfg.yang
Applying transform no_comments to Cisco-IOS-XR-um-vrf-cfg.yang.orig
Writing Cisco-IOS-XR-um-vrf-cfg.yang
Module tailf-common rev None not found in compilation context
Module tailf-common rev None not found in compilation context
Removing "ordered-by user" from /jc:configuration/protocols/bgp/group/neighbor
Removing "ordered-by user" from /jc:configuration/protocols/mpls/interface
Generating layer 2
+ Layer 2 adata unchanged
Generating base & TTT(?) for layer 2
+ Layer 2 base unchanged
+ Layer 2 TTT(?) unchanged
+ Layer 2 loose adata unchanged
Generating layer 1
+ Layer 1 adata changed
Generating base & TTT(?) for layer 1
+ Layer 1 base changed
+ Layer 1 TTT(?) changed
+ Layer 1 loose adata changed
Generating layer 0
+ Layer 0 adata unchanged
Generating base & TTT(?) for layer 0
+ Layer 0 base unchanged
+ Layer 0 TTT(?) unchanged
+ Layer 0 loose adata unchanged
Generating device type CiscoIosXr_24_1_ncs55a1
+ Device type CiscoIosXr_24_1_ncs55a1 adata unchanged
Generating device type JuniperCRPD_23_4R1_9
+ Device type JuniperCRPD_23_4R1_9 adata unchanged
+ Sys spec unchanged
Generating app layers.act
+ App layers.act unchanged

one-fabric-orch on  main [!?] via ↯ took 21s
```

But build failed:
```
❯ make build
acton build
Building project in /Users/klambrec/Developer/autlan/one-fabric-orch
Resolving dependencies (fetching if missing)...
Using cached pkg dependency actmf (N-V-__8AAJtaBABOb7WyjNcnWG3420pPMwvehI80qJohWsmC)
Using cached pkg dependency netconf (N-V-__8AAOL3AQD7Oy0qj2nAbYZ0u3dGWSGfvqABuALB3K0y)
Using cached pkg dependency orchestron (N-V-__8AAFBUEQBjXaW-92KIz_iNxo4_QeS9zuIj8QyUujZO)
Using cached pkg dependency rtrsecrets (N-V-__8AADYUAAD8XrwSuK5_7k2ua6Z05J5u-UIcoEhLNsmh)
Using cached pkg dependency yang (N-V-__8AAIOYIABUmXlnf1AZF-mtRvZcsWOi3MzODQhXHGxf)
[error Compilation error]: Hash info missing for one_fabric.layers.y_1.orchestron_rfs__rfs__base_config_entry
     ╭──▶ one_fabric/rfs.act@1:1-1:2
     │
   1 │ <no line>
     • ┬
     • ╰╸ Hash info missing for one_fabric.layers.y_1.orchestron_rfs__rfs__base_config_entry
─────╯
make: *** [build] Error 1
```

### Expected Behavior

Seammless adoption of the new YANG schema. This was resolved with `rm -rf out/`, so I take it the `.ty` file hashing failed to pick up the change?

--- a/compiler/acton/test_incremental.hs
+++ b/compiler/acton/test_incremental.hs
@@ -1612,6 +1612,38 @@ p35_changed_path_keeps_unaffected_provider =
     assertBool "expected provider for changed import a" (M.member (A.modName ["a"]) bProviders)
     assertBool "expected provider for unchanged import c" (M.member (A.modName ["c"]) bProviders)
 
+p36_removed_dep_name_triggers_front_refresh :: TestTree
+p36_removed_dep_name_triggers_front_refresh =
+  testCase "36-removed dep name forces front refresh" $ do
+    let proj = casesProjDir
+        src = casesSrcDir
+    ensureCasesProjectWithDeps [("libfoo", "deps/libfoo")]
+    depDir <- ensureDepProject proj "libfoo"
+    let depSrc = depDir </> "src" </> "libfoo.act"
+    writeFileUtf8 depSrc $ T.unlines
+      [ "def foo() -> int:"
+      , "    return 1"
+      ]
+    writeFileUtf8 (src </> "main.act") $ T.unlines
+      [ "import libfoo"
+      , ""
+      , "actor main(env: Env):"
+      , "    print(libfoo.foo())"
+      , "    env.exit(0)"
+      ]
+    res1 <- runActonIn proj ["build", "--color", "never", "--skip-build"]
+    assertExitSuccess "initial build" res1
+    writeFileUtf8 depSrc $ T.unlines
+      [ "def bar() -> int:"
+      , "    return 2"
+      ]
+    res2@(_ec2, out2) <- runActonIn proj ["build", "--color", "never", "--verbose", "--skip-build"]
+    assertExitFailure "rebuild after removing imported dep name" 1 res2
+    assertBool "expected stale-cache log for missing dep hash"
+      (T.isInfixOf "missing dep hashes in" out2 && T.isInfixOf "libfoo.foo" out2)
+    assertBool "did not expect internal hash-missing diagnostic"
+      (not (T.isInfixOf "Hash info missing for libfoo.foo" out2))
+
 -- Main -----------------------------------------------------------------------
 
 -- | Tasty entry point for incremental tests.
@@ -1667,5 +1699,6 @@ main = defaultMain $ localOption (NumThreads 1) $ testGroup "incremental"
       , p33_comprehensive_hashes
       , p34_removed_import_module
       , p35_changed_path_keeps_unaffected_provider
+      , p36_removed_dep_name_triggers_front_refresh
       ]
   ]

--- a/compiler/lib/src/Acton/Compile.hs
+++ b/compiler/lib/src/Acton/Compile.hs
@@ -2051,12 +2051,41 @@ compileTasks sp gopts opts rootPaths rootProj tasks callbacks = do
               return (fmap (\vals -> (n, vals)) resolvedQns)) (M.toList deps)
             return (fmap M.fromList resolved)
 
-          checkDeps label getHash users deps = do
+          -- For stale checks on cached .ty tasks we treat missing names/hashes
+          -- as "stale cache" signals and force front passes, instead of failing
+          -- early with internal hash diagnostics.
+          resolveQNameHashForStaleCheck getHash qn =
+            case qn of
+              A.GName m n -> resolveNameHashMap' m >>= \hm ->
+                return $ case hm of
+                  Nothing -> Left (missingIfaceDiagnostics mn "" m)
+                  Just hmap ->
+                    case M.lookup n hmap of
+                      Nothing -> Right Nothing
+                      Just info ->
+                        let h = getHash info
+                        in if B.null h then Right Nothing else Right (Just h)
+              A.QName m n -> resolveNameHashMap' m >>= \hm ->
+                return $ case hm of
+                  Nothing -> Left (missingIfaceDiagnostics mn "" m)
+                  Just hmap ->
+                    case M.lookup n hmap of
+                      Nothing -> Right Nothing
+                      Just info ->
+                        let h = getHash info
+                        in if B.null h then Right Nothing else Right (Just h)
+              A.NoQ _ -> return (Right Nothing)
+
+          checkDeps _label getHash _users deps = do
             resolved <- traverseDiags (\(qn, recorded) -> do
-              let userNote = fmtUsers users qn
-              currE <- resolveQNameHash label getHash userNote qn
+              currE <- resolveQNameHashForStaleCheck getHash qn
               return (fmap (\curr -> (qn, recorded, curr)) currE)) (M.toList deps)
-            return (fmap (\triples -> [ (qn, old, new) | (qn, old, new) <- triples, old /= new ]) resolved)
+            return (fmap
+              (\triples ->
+                let deltas = [ (qn, old, new) | (qn, old, Just new) <- triples, old /= new ]
+                    missing = [ qn | (qn, _old, Nothing) <- triples ]
+                in (deltas, missing))
+              resolved)
 
       case gtTask t of
         ParseErrorTask{ parseDiagnostics = diags } -> return (key, Left diags)
@@ -2096,22 +2125,23 @@ compileTasks sp gopts opts rootPaths rootProj tasks callbacks = do
                   case (pubRes, implRes) of
                     (Left diags, _) -> return (Left diags)
                     (_, Left diags) -> return (Left diags)
-                    (Right pubDeltas, Right implDeltas) ->
-                      return (Right (pubDeltas, implDeltas, pubUsers, implUsers))
+                    (Right (pubDeltas, pubMissing), Right (implDeltas, implMissing)) ->
+                      return (Right (pubDeltas, implDeltas, pubMissing, implMissing, pubUsers, implUsers))
             -- Source tasks always run front passes, so deps are irrelevant.
-            _ -> return (Right ([], [], M.empty, M.empty))
+            _ -> return (Right ([], [], [], [], M.empty, M.empty))
 
           case needByDepsRes of
             Left diags -> return (key, Left diags)
-            Right (pubDeltas, implDeltas, pubUsers, implUsers) -> do
+            Right (pubDeltas, implDeltas, pubMissing, implMissing, pubUsers, implUsers) -> do
               let needBySource = case gtTask t of { ActonTask{} -> True; _ -> False }
                   -- Public deltas require front passes; impl deltas only need back jobs.
                   needByPub = not (null pubDeltas)
+                  needByMissing = not (null pubMissing) || not (null implMissing)
                   needByImpl = not (null implDeltas)
                   forceAlt    = altOutput optsT && mn == rootAlt
                   forceAlways = C.alwaysbuild optsT
                   -- Front passes run on source or API changes, or when forced.
-                  needFront = needBySource || needByPub || forceAlt || forceAlways
+                  needFront = needBySource || needByPub || needByMissing || forceAlt || forceAlways
                   mModuleImplHash = case gtTask t of
                     TyTask{ tyImplHash = implHash } -> Just implHash
                     _ -> Nothing
@@ -2129,9 +2159,21 @@ compileTasks sp gopts opts rootPaths rootProj tasks callbacks = do
                     when (C.verbose gopts) $ do
                       if needBySource
                         then ccOnInfo callbacks ("  Stale " ++ modNameToString mn ++ ": source changed")
-                        else when needByPub $ do
-                          let fmtDelta (qn, old, new) = prstr qn ++ " " ++ short8 old ++ " → " ++ short8 new ++ fmtUsers pubUsers qn
-                          ccOnInfo callbacks ("  Stale " ++ modNameToString mn ++ ": pub changes in " ++ Data.List.intercalate ", " (map fmtDelta pubDeltas))
+                        else do
+                          when needByPub $ do
+                            let fmtDelta (qn, old, new) = prstr qn ++ " " ++ short8 old ++ " → " ++ short8 new ++ fmtUsers pubUsers qn
+                            ccOnInfo callbacks ("  Stale " ++ modNameToString mn ++ ": pub changes in " ++ Data.List.intercalate ", " (map fmtDelta pubDeltas))
+                          when needByMissing $ do
+                            let fmtMissing users qn = prstr qn ++ fmtUsers users qn
+                                pubMissingItems =
+                                  [ "pub " ++ fmtMissing pubUsers qn
+                                  | qn <- Data.List.sortOn Hashing.qnameKey pubMissing
+                                  ]
+                                implMissingItems =
+                                  [ "impl " ++ fmtMissing implUsers qn
+                                  | qn <- Data.List.sortOn Hashing.qnameKey implMissing
+                                  ]
+                            ccOnInfo callbacks ("  Stale " ++ modNameToString mn ++ ": missing dep hashes in " ++ Data.List.intercalate ", " (pubMissingItems ++ implMissingItems))
                     t' <- case gtTask t of
                             ActonTask{} -> return (gtTask t)
                             TyTask{}    -> do

